docs/designs/0000-template.md

```diff
@@ -0,0 +1,23 @@
+# Title
+
+Updated by: [0000-this-doc-does-not-exist]()
+
+Obsoleted by: [0000-this-doc-does-not-exist]()
+
+*(these sections should only be added when necessary, and should always include the links to the relevant documents)*
+
+## Introduction
+
+This section should bring a short summary of what this document is covering, the main points it is going to describe, and why it is important. Ideally should be kept under two paragraphs.
+
+## Context [optional]
+
+Explains the context outside the mere technical aspects of the design that existed when it was written. It must make clear what other influences where at play that affected the decisions that were taken. In case there is nothing relevant to mention about the context, this section can be skipped.
+
+## [Free-form sections]
```

Comment:
I responded to the Go sample, I don't think this main H2 section should be free-form, its subsections sure can be.

Reply:
Let's concentrate this discussion in this [thread](https://github.com/containerbuildsystem/cachi2/pull/601#discussion_r1724627700).